pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -715,20 +715,11 @@ func removeTerminatedPods(pods []*runtimeapi.PodSandbox) []*runtimeapi.PodSandbo
 
 	result := make([]*runtimeapi.PodSandbox, 0)
 	for _, refs := range podMap {
-		if len(refs) == 1 {
```

Comment:
it doesn't make sense to me what this original code block was intending to do....

we sort pods, put them in a map, iterate the map, and filter out pods that were not ready... why this particular check existed makes no sense to me.

Reply:
looking closer at the original comment, "It only removes a pod when there is a running instance of a pod in same name or namespace" is unclear to me as well.